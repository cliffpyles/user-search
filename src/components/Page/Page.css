.Page {
  display: grid;
  grid-template-columns: 8;
  grid-template-rows: auto;
  grid-template-areas:
    'PageHeader PageHeader PageHeader PageHeader PageHeader PageHeader PageHeader PageHeader'
    'PageContent PageContent PageContent PageContent PageContent PageContent PageSidebar PageSidebar'
    'PageFooter PageFooter PageFooter PageFooter PageFooter PageFooter PageFooter PageFooter';
}

.PageContent {
  grid-area: PageContent;
}

.PageFooter {
  grid-area: PageFooter;
}

.PageHeader {
  grid-area: PageHeader;
}

.PageSidebar {
  grid-area: PageSidebar;
}
