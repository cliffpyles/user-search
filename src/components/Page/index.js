export {
  default,
  PageContent,
  PageFooter,
  PageHeader,
  PageSidebar
} from './Page'
