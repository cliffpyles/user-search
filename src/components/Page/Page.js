import React from 'react'
import './Page.css'

export default function Page({ children }) {
  return <div className="Page">{children}</div>
}

export const PageContent = ({ children }) => {
  return <div className="PageContent">{children}</div>
}

export const PageFooter = ({ children }) => {
  return <div className="PageFooter">{children}</div>
}

export const PageHeader = ({ children }) => {
  return <div className="PageHeader">{children}</div>
}

export const PageSidebar = ({ children }) => {
  return <div className="PageSidebar">{children}</div>
}
