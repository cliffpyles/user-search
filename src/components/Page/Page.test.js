import React from 'react'
import Renderer from 'react-test-renderer'
import Page from './'

test('can render a Page', () => {
  const component = Renderer.create(
    <Page>
      <p>A little content for the page.</p>
    </Page>
  )

  expect(component.toJSON()).toMatchSnapshot()
})

test('can render a Page with title', () => {
  const component = Renderer.create(
    <Page title="Example Page">
      <p>A little content for the page.</p>
    </Page>
  )

  expect(component.toJSON()).toMatchSnapshot()
})
