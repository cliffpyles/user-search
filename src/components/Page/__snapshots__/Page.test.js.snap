// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a Page 1`] = `
<div
  className="Page"
>
  <p>
    A little content for the page.
  </p>
</div>
`;

exports[`can render a Page with title 1`] = `
<div
  className="Page"
>
  <p>
    A little content for the page.
  </p>
</div>
`;
